tswow-core/Private/TSItem.cpp

```diff
@@ -690,6 +690,24 @@ TSItemTemplate TSItem::GetTemplate()
     return TSItemTemplate(item->GetTemplate());
 }
 
+struct VirtualItemTemplate : ItemTemplate
+{
+    VirtualItemTemplate(ItemTemplate const* base) : ItemTemplate(*base), seed(base->ItemId)
+    {
+    }
+
+    /**
+     * stat seed??
+     */
+    uint32 seed;
+
+};
+
+TSItemTemplate TSItem::GetTemplateCopy()
+{
+    return TSItemTemplate(new VirtualItemTemplate(item->GetTemplate()));
```

Comment:
I grabbed this from Foe, it allows for a deep copy of the item but I am unsure if this would still be removed when the item itself would be deleted. Would need to look into that 1

Reply:
The constructor would still work if used as a smart pointer, but `TSItemTemplate` will not clean up the stray itemtemplate, so that's a memory leak. Normally, ItemTemplates are owned by the ObjectMgr since it's just static data, but with dynamic data we do need to consider where it should live, _or_ use a smart pointer. I think the latter is suitable here since it'd be used very similarly to TSOutfits/dynamic dressnpcs

Reply:
and i assume `Item` objects would need the ability to hold those smart pointers, so they stay loaded for the items they represent

Reply:
If I merge these 2 functions together, I could utilize the data and then delete the object. or just make SaveItemTemplate as should always be run last. unsure if there is a time where you would want the save and not the reload of the objmgr to apply values to the player.
SaveItemTemplate()
ReloadSingleItemTemplateObject(TSItemTemplate)

Reply:
all of this kinda got redone in v2, so this conversation shouldnt matter as much. perhaps still a mem issue? but i think as it's only used and then cast away it's probably fine. It only allows this once per itemID anyways, so worst case it is a not horrible leak compared to before

Reply:
The general rule of thumb is that for anything you allocate with "new", you also need a way to "delete" it somewhere else. Storing things temporarily on the heap is fine, as long as it's deleted.

Reply:
Well on the new system it is now handled by 
![image](https://user-images.githubusercontent.com/42619070/161059422-ecd5d3d5-3f8c-4925-9a60-220db90668b4.png)
and 
![image](https://user-images.githubusercontent.com/42619070/161059470-ed80b4d3-ef11-428c-a547-204b8516bdd8.png)

Where after it is used, it now goes over to the objmgr directly where the core itself would handle it like a normal item_template so the serverside should delete it on it's own whenever it would be necessary

Reply:
This still leaks. You do `_itemTemplateStore[itemInfo->ItemId] = *itemInfo`, which creates a copy of your heap structure to store in the item template store, but does not track the heap pointer itself which is lost when that function goes out of scope. 

You don't really need to put anything on the heap here, just create an overload for `ObjectMgr::LoadSingleItemTemplateObject` to accept the old and new id, create a new VirtualItemTemplate (accept both the old object and new id in the constructor and call the super constructor / change entry from there instead of wrapping it in a TSItemTemplate for that) and store it directly inside the `_itemTemplateStore` by value (this avoids both the heap and unnecessary copies)

Reply:
new commit fixing mem issues; [bfd444b](https://github.com/tswow/tswow/pull/472/commits/bfd444b9845bf07c6c7ee9f50f395abac2d1ff46)